packages/mermaid/src/diagrams/git/gitGraphRenderer.js

```diff
@@ -388,49 +391,59 @@ const findLane = (y1, y2, depth = 0) => {
  * Draw the lines between the commits. They were arrows initially.
  *
  * @param {any} svg
- * @param {any} commit1
- * @param {any} commit2
+ * @param {any} commitA
+ * @param {any} commitB
  * @param {any} allCommits
  */
-const drawArrow = (svg, commit1, commit2, allCommits) => {
-  const p1 = commitPos[commit1.id];
-  const p2 = commitPos[commit2.id];
-  const overlappingCommits = hasOverlappingCommits(commit1, commit2, allCommits);
-  // log.debug('drawArrow', p1, p2, overlappingCommits, commit1.id, commit2.id);
+const drawArrow = (svg, commitA, commitB, allCommits) => {
+  const p1 = commitPos[commitA.id];
+  const p2 = commitPos[commitB.id];
+  const arrowNeedsRerouting = shouldRerouteArrow(commitA, commitB, allCommits);
+  // log.debug('drawArrow', p1, p2, arrowNeedsReroute, commitA.id, commitB.id);
 
   let arc = '';
   let arc2 = '';
   let radius = 0;
   let offset = 0;
-  let colorClassNum = branchPos[commit2.branch].index;
+  let colorClassNum = branchPos[commitB.branch].index;
   let lineDef;
-  if (overlappingCommits) {
+  if (arrowNeedsRerouting) {
     arc = 'A 10 10, 0, 0, 0,';
     arc2 = 'A 10 10, 0, 0, 1,';
     radius = 10;
     offset = 10;
-    // Figure out the color of the arrow,arrows going down take the color from the destination branch
-    colorClassNum = branchPos[commit2.branch].index;
 
     const lineY = p1.y < p2.y ? findLane(p1.y, p2.y) : findLane(p2.y, p1.y);
     const lineX = p1.x < p2.x ? findLane(p1.x, p2.x) : findLane(p2.x, p1.x);
 
     if (dir === 'TB') {
       if (p1.x < p2.x) {
+        // Destination commit is on branch position left of source commit
+        // so render arrow leftward with colour of destination branch
+        colorClassNum = branchPos[commitB.branch].index;
```

Comment:
We have to cover this lines by tests

Reply:
Thanks @nirname. I can see [lines 423-424](https://app.codecov.io/gh/mermaid-js/mermaid/pull/4927/blob/packages/mermaid/src/diagrams/git/gitGraphRenderer.js#L423) and [lines 439-440](https://app.codecov.io/gh/mermaid-js/mermaid/pull/4927/blob/packages/mermaid/src/diagrams/git/gitGraphRenderer.js#L439) are not covered so I'll write tests for those. (I thought they would be covered already because the conditions are not new. When I write the new tests, they should increase coverage overall.)

I can see other areas of the file still need coverage but I think these should be a separate PR/issue so I'll leave them for now. Let me know if you think I've missed any though.

Reply:
No need to meet all the demands of code analyzers in one approach. Should we need to fix everything we could before applying other incoming improvements the development would stop. Focus on those lines that are the heart of this change. Don't do extra job.

Reply:
@nirname I think I've resolved this issue now (I hope). There's info in the commit messages about reasoning. I've also increased e2e test coverage slightly.